cmd/wallet-sdk-gomobile/wrapper/httpclient.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wrapper
+
+import (
+	"crypto/tls"
+	"net/http"
+
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+)
+
+// HTTPClient is the HTTP client implementation that gets injected in from the gomobile layer.
+// It makes use of the optional HTTP client parameters that are exposed from the various APIs.
+type HTTPClient struct {
+	additionalHeaders      []api.Header
+	disableTLSVerification bool
+}
+
+// NewHTTPClient returns a new HTTPClient.
+func NewHTTPClient() *HTTPClient {
+	return &HTTPClient{}
+}
+
+// AddHeaders adds the given headers to the list of headers that will get added to all HTTP calls made by this
+// client implementation.
+func (m *HTTPClient) AddHeaders(headers *api.Headers) {
+	m.additionalHeaders = append(m.additionalHeaders, headers.GetAll()...)
+}
+
+// DisableTLSVerification disables TLS verification. This should only be used for testing purposes.
+func (m *HTTPClient) DisableTLSVerification() {
+	m.disableTLSVerification = true
+}
+
+// Do sends the given HTTP request and returns an HTTP response. If this HTTP client has any additional headers to be
+// injected (from previous calls to AddHeaders), then those headers are added to the request before being executed
+// by the default Go HTTP client.
+func (m *HTTPClient) Do(req *http.Request) (*http.Response, error) {
+	for _, additionalHeader := range m.additionalHeaders {
+		req.Header.Add(additionalHeader.Name, additionalHeader.Value)
+	}
+
+	if m.disableTLSVerification {
```

Comment:
In this PR, I have the split between defining a custom HTTP client and TLS config vs. using http.DefaultClient, but I think the two approaches can be "merged". However, from what I was reading about, there are some considerations about defining your own client like we have below. In my next PR I'll be adding support for passing in a timeout, and I'll also figure out the best way to merge the two approaches to make this code cleaner.